CourseGrab/Home/HomeTableViewController.swift

```diff
@@ -114,6 +66,41 @@ class HomeTableViewController: UITableViewController {
 
 }
 
+// MARK: - Networking
+
+extension HomeTableViewController {
+
+    private func getAllTrackedCourses() {
+        NetworkManager.shared.getAllTrackedCourses().observe { result in
+            switch result {
+            case .value(let response):
+                let available = response.data.filter { $0.status == .open }
+                if !available.isEmpty {
+                    self.tableSections.append(.available(available))
+                }
+                let awaiting = response.data.filter { $0.status != .open }
+                if !awaiting.isEmpty {
+                    self.tableSections.append(.awaiting(awaiting))
+                }
+                var newSections: IndexSet = []
+                (0..<self.tableSections.count).forEach { newSections.insert($0) }
```

Comment:
you could do `let newSections = (0..<self.tableSections.count).map { $0 }`? does `insert` place it in  index 0?

Reply:
oh gotcha